extra/release_tool.py

```diff
@@ -2389,8 +2400,11 @@ def do_release(release_state_file):
     if input.startswith("Y") or input.startswith("y"):
         refresh_repos(state)
 
-    for repo in sorted(Component.get_components_of_type("git"), key=repo_sort_key):
-        determine_version_to_include_in_release(state, repo)
+    repos = sorted(Component.get_components_of_type("git"), key=repo_sort_key)
+    while len(repos) > 0:
+        repo = repos.pop(0)
+        if not determine_version_to_include_in_release(state, repo):
+            repos.append(repo)
```

Comment:
Clever! I had to read this a few times to understand that it keeps looping until all entries are decided.